### DISCUSSION TITLE
Ability to login to directus using access token of a configured provider.

### DISCUSSION THREAD

**Discussion by useEffects** [directus/directus#24299] [Feature Requests]
### Summary

Access token of the provider can be generated by using third party libraries and can the auth can be maintained by it. Through this new feature, we can exchange access token of the provider to the access token of directus and create a session to consume the resources in directus. 

### Basic Example

A `router.post` method in https://github.com/directus/directus/blob/9cbd17f1df60da27c5feba88883e357e1bbd03a3/api/src/auth/drivers/openid.ts#L378

preferably, the post method should take `access_token` in the body.

### Motivation

Directus has the ability to configure multiple auth providers to login. As of now, the only way to login into directus without seeing the directus login page is to use the seamless SSO approach mentioned in the docs: https://docs.directus.io/self-hosted/sso.html#seamless-sso.  Unfortunately, there is no option for a machine to machine way of making the user logging in. There could be scenarios such as the user is logged in with keycloak already, now they want to login to directus using the access token generated by keycloak.

### Detailed Design

lorem ipsum

### Must Have:
- Ability to authenticate on directus sdk with auto token refresh and websocket features (complete features).
-  Ability to be able to login using POST requests
- Ability to be able to login using graphql

### Drawbacks

lorem ipsum

### Alternatives

Currently, we have to build our own custom endpoint in directus, manually verify the token, create a directus signed token and return as response. This will work even if the provider is not configured in directus. There should be a better approach for this.

### Adoption Strategy

Not a breaking change, no need of migration.

**Comment by useEffects** [comment]
btw, there is no `POST /auth/login/:provider` as mentioned in the docs: https://docs.directus.io/reference/authentication.html#login. There is only GET request for the same

**Reply by BDEERAJ** [reply]
Can you suggest any free resources to learn frontend development? There are very few good resources on YouTube, and I need something better. I'm asking because your response was uploaded 3 hours ago
